File: admin-panel/convert.py
import os
from bs4 import BeautifulSoup
import re

def html_to_jsx(html):
    # Basic replacements
    jsx = html.replace('class="', 'className="')
    jsx = jsx.replace('for="', 'htmlFor="')
    
    # Close img and input tags
    jsx = re.sub(r'(<img[^>]*?)(?<!/)>', r'\1 />', jsx)
    jsx = re.sub(r'(<input[^>]*?)(?<!/)>', r'\1 />', jsx)
    jsx = re.sub(r'(<br[^>]*?)(?<!/)>', r'\1 />', jsx)
    jsx = re.sub(r'(<hr[^>]*?)(?<!/)>', r'\1 />', jsx)
    
    # Inline styles
    def style_replacer(match):
        style_str = match.group(1)
        # simplistic conversion for standard styles like "font-variation-settings: 'FILL' 1;"
        rules = style_str.split(';')
        react_styles = []
        for rule in rules:
            if not rule.strip(): continue
            parts = rule.split(':', 1)
            if len(parts) == 2:
                key = parts[0].strip()
                val = parts[1].strip()
                # camelCase key
                key = re.sub(r'-([a-z])', lambda m: m.group(1).upper(), key)
                react_styles.append(f'{key}: "{val}"')
        return 'style={{' + ', '.join(react_styles) + '}}'
        
    jsx = re.sub(r'style="([^"]*)"', style_replacer, jsx)
    
    # Replace html comments
    jsx = re.sub(r'<!--(.*?)-->', r'{/*\1*/}', jsx, flags=re.DOTALL)
    
    return jsx

configs = [
    {
        "src": r"D:\Stay Q\stitch_export\stitch_stay_q_command_center\booking_ledger\code.html",
        "dest": r"d:\Stay Q\admin-panel\src\app\bookings\page.tsx",
        "name": "BookingsPage",
        "api": "/api/v1/admin/bookings"
    },
    {
        "src": r"D:\Stay Q\stitch_export\stitch_stay_q_command_center\people_management\code.html",
        "dest": r"d:\Stay Q\admin-panel\src\app\people\page.tsx",
        "name": "PeoplePage",
        "api": "/api/v1/admin/users"
    },
    {
        "src": r"D:\Stay Q\stitch_export\stitch_stay_q_command_center\system_health\code.html",
        "dest": r"d:\Stay Q\admin-panel\src\app\health\page.tsx",
        "name": "HealthPage",
        "api": "/api/v1/admin/health"
    },
    {
        "src": r"D:\Stay Q\stitch_export\stitch_stay_q_command_center\dashboard_overview\code.html",
        "dest": r"d:\Stay Q\admin-panel\src\app\executive\page.tsx",
        "name": "ExecutivePage",
        "api": "/api/v1/admin/analytics/overview"
    }
]

for cfg in configs:
    with open(cfg["src"], "r", encoding="utf-8") as f:
        soup = BeautifulSoup(f.read(), "html.parser")
        
    main_tag = soup.find("main")
    if not main_tag:
        # Some might not have <main>, find the main container instead.
        # But looking at the HTML, all of them have <main> except maybe dashboard if it's different.
        main_tag = soup.find("body") # fallback
        
    # extract innerHTML of the tag, or just the tag itself. We'll take the tag itself.
    html_content = str(main_tag)
    
    # Convert to JSX
    jsx_content = html_to_jsx(html_content)
    
    # Generate Page component
    page_code = f"""\"use client\";

import React, {{ useEffect, useState }} from 'react';
import axios from 'axios';

export default function {cfg['name']}() {{
  const [data, setData] = useState<any>(null);

  useEffect(() => {{
    axios.get('{cfg['api']}')
      .then(res => setData(res.data))
      .catch(err => console.error(err));
  }}, []);

  return (
    <>
      {{/* data points can be wired below using data?.field etc. */}}
      {jsx_content}
    </>
  );
}}
"""
    os.makedirs(os.path.dirname(cfg["dest"]), exist_ok=True)
    with open(cfg["dest"], "w", encoding="utf-8") as f:
        f.write(page_code)
    print(f"Generated {cfg['dest']}")

File: admin-panel/src/app/convert.js
const fs = require('fs');
const path = require('path');

const screens = [
    { name: 'business_reports', out: 'reports/page.tsx', endpoint: '/api/v1/admin/reports' },
    { name: 'revenue_management', out: 'revenue/page.tsx', endpoint: '/api/v1/admin/revenue' },
    { name: 'multi_currency_hub', out: 'currency/page.tsx', endpoint: '/api/v1/admin/currencies' },
    { name: 'tax_compliance', out: 'taxes/page.tsx', endpoint: '/api/v1/admin/taxes' }
];

const baseSrc = 'D:/Stay Q/stitch_export/stitch_stay_q_command_center';
const baseDest = 'd:/Stay Q/admin-panel/src/app';

screens.forEach(s => {
    let html = fs.readFileSync(path.join(baseSrc, s.name, 'code.html'), 'utf-8');
    
    // Extract main and everything after it (like modals) but before </body>
    const mainStart = html.indexOf('<main');
    const bodyEnd = html.indexOf('</body>');
    let content = html.slice(mainStart, bodyEnd);
    
    // Remove header
    content = content.replace(/<header[\s\S]*?<\/header>/i, '');
    
    // Convert to JSX
    content = content.replace(/class=/g, 'className=');
    content = content.replace(/onclick=/g, 'onClick=');
    content = content.replace(/for=/g, 'htmlFor=');
    content = content.replace(/checked=\"\"/g, 'defaultChecked');
    content = content.replace(/disabled=\"\"/g, 'disabled');
    
    // Self-close tags
    content = content.replace(/<input([^>]*[^\/])>/g, '<input$1 />');
    content = content.replace(/<img([^>]*[^\/])>/g, '<img$1 />');
    content = content.replace(/<hr([^>]*[^\/])>/g, '<hr$1 />');
    content = content.replace(/<br([^>]*[^\/])>/g, '<br$1 />');
    
    // specific styles
    content = content.replace(/style=\"([^\"]*)\"/g, (match, p1) => {
        let styleObj = {};
        p1.split(';').forEach(rule => {
            let parts = rule.split(':');
            if (parts.length > 1) {
                let key = parts[0].trim().replace(/-([a-z])/g, g => g[1].toUpperCase());
                let val = parts[1].trim();
                styleObj[key] = val.replace(/'/g, "");
            }
        });
        return `style={${JSON.stringify(styleObj)}}`;
    });
    
    // Fix viewBox, stroke-width, stroke-linecap, stroke-dasharray, stroke-dashoffset, preserveAspectRatio, linearGradient
    content = content.replace(/viewbox/g, 'viewBox');
    content = content.replace(/stroke-width/g, 'strokeWidth');
    content = content.replace(/stroke-linecap/g, 'strokeLinecap');
    content = content.replace(/stroke-dasharray/g, 'strokeDasharray');
    content = content.replace(/stroke-dashoffset/g, 'strokeDashoffset');
    content = content.replace(/preserveaspectratio/g, 'preserveAspectRatio');
    content = content.replace(/lineargradient/g, 'linearGradient');

    // Fix raw JS in onClick (just mock it)
    content = content.replace(/onClick=\"[^\"]*\"/g, 'onClick={() => {}}');
    
    // Fix <!-- --> comments (remove them)
    content = content.replace(/<!--[\s\S]*?-->/g, '');

    const jsx = `"use client";
import React, { useEffect, useState } from 'react';
import axios from 'axios';

export default function Page() {
    const [data, setData] = useState<any>(null);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        axios.get('${s.endpoint}')
            .then(res => {
                setData(res.data);
                setLoading(false);
            })
            .catch(err => {
                console.error(err);
                setLoading(false);
            });
    }, []);

    return (
        <>
            {/* You can use {data?.something || 'Fallback'} to render API data */}
            ${content}
        </>
    );
}
`;

    const destDir = path.join(baseDest, path.dirname(s.out));
    fs.mkdirSync(destDir, { recursive: true });
    fs.writeFileSync(path.join(baseDest, s.out), jsx);
});
console.log('Done!');
